lib/Unexpected.js

```diff
@@ -484,14 +482,30 @@ Unexpected.prototype.fail = function (arg) {
                 output.error('Explicit failure');
             }
         };
+        var expect = this.expect;
         Object.keys(arg).forEach(function (key) {
             var value = arg[key];
             if (key === 'diff') {
-                error.createDiff = value;
+                if (typeof value === 'function' && this.parent) {
+                    error.createDiff = function (output, diff, inspect, equal) {
+                        var inheritedOutput = expect.createOutput(output.format);
+                        inheritedOutput.inline = output.inline;
+                        inheritedOutput.output = output.output;
```

Comment:
Turns out you can, and (at least) unexpected-messy depends on it, so that's why this had to go in :(

I actually wanted to raise exactly that discussion because I felt the same way as you -- as of the next major release I think we should require both `inspect` and `diff` to return the output and not allow using them to add to an existing pen. That'll allow us to get rid of a lot of the hacks here and elsewhere.